pkg/gardenlet/bootstrappers/verify_version_test.go

```diff
@@ -0,0 +1,95 @@
+			Entry("fail because gardenlet version is too low (both version suffixed with '-dev')", "v1.3.3-dev", "v1.0.4-dev", MatchError(ContainSubstring("gardenlet version must not be older than two minor gardener-apiserver versions"))),
+
+			Entry("succeed because gardenlet version is equal to gardener-apiserver version", "v1.2.3", "v1.2.3", Succeed()),
+			Entry("succeed because gardenlet version is equal to gardener-apiserver version (gardener-apiserver version suffixed with '-dev')", "v1.2.3-dev", "v1.2.3", Succeed()),
+			Entry("succeed because gardenlet version is equal to gardener-apiserver version (gardenlet version suffixed with '-dev')", "v1.2.3", "v1.2.3-dev", Succeed()),
+			Entry("succeed because gardenlet version is equal to gardener-apiserver version (both versions suffixed with '-dev')", "v1.2.3-dev", "v1.2.3-dev", Succeed()),
+
+			Entry("succeed because gardenlet patch version is lower than gardener-apiserver version", "v1.2.3", "v1.2.2", Succeed()),
+			Entry("succeed because gardenlet patch version is lower than gardener-apiserver version (gardener-apiserver version suffixed with '-dev')", "v1.2.3-dev", "v1.2.2", Succeed()),
+			Entry("succeed because gardenlet patch version is lower than gardener-apiserver version (gardenlet version suffixed with '-dev')", "v1.2.3", "v1.2.2-dev", Succeed()),
+			Entry("succeed because gardenlet patch version is lower than gardener-apiserver version (both versions suffixed with '-dev')", "v1.2.3-dev", "v1.2.2-dev", Succeed()),
+
+			Entry("succeed because gardenlet minor version is lower (by 1) than gardener-apiserver version", "v1.2.3", "v1.1.3", Succeed()),
+			Entry("succeed because gardenlet minor version is lower (by 1) than gardener-apiserver version (gardener-apiserver version suffixed with '-dev')", "v1.2.3-dev", "v1.1.3", Succeed()),
+			Entry("succeed because gardenlet minor version is lower (by 1) than gardener-apiserver version (gardenlet version suffixed with '-dev')", "v1.2.3", "v1.1.3-dev", Succeed()),
+			Entry("succeed because gardenlet minor version is lower (by 1) than gardener-apiserver version (both versions suffixed with '-dev')", "v1.2.3-dev", "v1.1.3-dev", Succeed()),
+
+			Entry("succeed because gardenlet minor version is lower (by 2) than gardener-apiserver version", "v1.2.3", "v1.0.3", Succeed()),
+			Entry("succeed because gardenlet minor version is lower (by 2) than gardener-apiserver version (gardener-apiserver version suffixed with '-dev')", "v1.2.3-dev", "v1.0.3", Succeed()),
+			Entry("succeed because gardenlet minor version is lower (by 2) than gardener-apiserver version (gardenlet version suffixed with '-dev')", "v1.2.3", "v1.0.3-dev", Succeed()),
+			Entry("succeed because gardenlet minor version is lower (by 2) than gardener-apiserver version (both versions suffixed with '-dev')", "v1.2.3-dev", "v1.0.3-dev", Succeed()),
```

Comment:
I do not understand how this should succeed when the code seemingly checks for a difference of one minor version. What am I missing?